world/stat_checks/check_maker.py

```diff
@@ -88,6 +94,82 @@ def announce_to_room(self):
             self.roll_message, options={"roll": True}
         )
 
+    def announce_to_players(self):
+        """
+        Sends a private roll result message to specific players as well as
+        all staff at self.character's location.
+        """
+        self_list = (
+            "me",
+            "self",
+            str(self.character).lower(),
+            str(self.character.key).lower(),
+        )
+
+        # If we fail to make a copy (e.g. self.receivers is None),
+        # start with a blank slate.
+        try:
+            receiver_list = self.receivers.copy()
+        except:
+            receiver_list = []
+
+        # Adjust receiver list; this is who actually will receive it besides staff and the caller.
+        # NOTE: receiver_names became necessary as .join() was being weird with converting
+        # str(self.receivers).  Results were N, a, m, e, 1 instead of Name1, Name2, etc.
+        receiver_names = []
+        for receiver in self.receivers:
+            # If I am the caller or a staff member, remove me from the list of receivers.
+            # (Staff and the caller always get the memo; we want PC receivers.)
+            # Also checks for duplicate receivers to cross them off too so malicious users
+            # can't put "=Name,Name,Name" and spam the other party with the result.
+            if (
+                receiver.check_permstring("Builders")
+                or receiver.name.lower() in self_list
+                or receiver_list.count(receiver) > 1
+            ):
+                receiver_list.remove(receiver)
+            else:
+                receiver_names.append(receiver.name)
```

Comment:
while this works there's simpler approaches. python has a `set()` built-in type that is unordered but has no duplicates. So an easy way to do this would be something like:
```
receiver_list = [ob for ob in set(self.receivers) if not ob.check_permstring("Builders") and ob.name.lower() not in self_list]
receiver_names = [ob.name for ob in receiver_list]
```